### ISSUE TITLE
[Bug] Issue in using relational db mysql and postgres as source for pulsar io topics

### ISSUE THREAD

**Issue by abhinavisin** [apache/pulsar#22083]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Ubuntu 18.04
Pulsar 3.0.1
Mysql - Server version: 5.7.25-log MySQL Community Server (GPL)
Postgres - 14.10

### Minimal reproduce step

Step 1. Follow the steps documented for bringing CDC connectors outlined in example 
https://pulsar.apache.org/docs/3.0.x/io-cdc-debezium/
            Docker run mysql instance 
Step 2. Start a standalone pulsar instance 
           bin/pulsar standalone
Step 3. Start the Pulsar Debezium connector in local run mode 
           Attached yaml file (debezium connector config) and logs of execution 
Step 4. No topics created for any table from "inventory" schema 
             Only offset and history topics are created in
[pulsar-debezium-mysql.log](https://github.com/apache/pulsar/files/14351115/pulsar-debezium-mysql.log)
 pulsar
[debezium-mysql-source-config.txt](https://github.com/apache/pulsar/files/14351137/debezium-mysql-source-config.txt)

Above was repeated for postgres and same results. 
**However if using pulsar 2.6.0 all topics and tables are created successfully.** 

### What did you expect to see?

No topic created for any table of the mysql and postgres in pulsar 3.0.1.
Only following topics created for mysql table- 
persistent://public/default/offset-topic
persistent://public/default/history-topic

### What did you see instead?

Subscribe to the topic sub-products for the inventory.products table and get change data from relational db.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
> Pulsar 3.0.1

Do you have a chance to test with Pulsar 3.0.2 since that's the latest released 3.0.x version. Just to make sure that it's not already fixed. Thanks!

**Comment by abhinavisin** [comment]
> > Pulsar 3.0.1
> 
> Do you have a chance to test with Pulsar 3.0.2 since that's the latest released 3.0.x version. Just to make sure that it's not already fixed. Thanks!

Yes I have tested with following versions 3.0.2, 3.0.1 and 2.9.0 all have the same results.
It appears architecture of debezium snapshot changed and the Producer is not getting invoked to create topics per table. 
While in pulsar 2.6.0 it embeds debzium older arch which is able to generate topics per table.

**Comment by abhinavisin** [comment]
Are there any automation/sanity tests for "source" operations per release to reference and compare if these examples are passing per release.
